hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```
@@ -148,13 +148,15 @@ public EncryptionManager encryption() {
     if (tableKeyId != null) {
       if (keyManagementClient == null) {
         throw new RuntimeException(
-            "Cant create encryption manager, because key management client is not set");
+            "Cannot create encryption manager without a key management client");
```

This is a nit, matching the "Cannot" language used elsewhere. (Not strongly opinionated though)

---

We could take this further with maybe:

```java
throw new RuntimeException(
    "Cannot create encryption manager without a key management client. Consider setting the '"
        + CatalogProperties.ENCRYPTION_KMS_IMPL + "' catalog property.");
```

However, this might be a bit premature

---

We could also throw a different exception class here

---

IllegalArgs or a Precondition prompting the key to be set sounds reasonable to me

---

+1 to use IllegalArgumentException or Preconditions check.